### PR TITLE
Remap tabs when open/close/activated #179

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

**Comment by jonathanKingston** [comment]
Code looks fine, have not tested etc but logic seems sound. @bakulf is the issue that at the time of the tab events like on open the usercontext-id isn't assigned to the tab?

Is there any certainty `this._getUserContextIdFromTab` actually fixes this issue or does it just increase the time between a race condition happening?

**Comment by jonathanKingston** [comment]
Yeah so this change actually just delays the call to `this._getUserContextIdFromTab`  but `restyleTab` in `tabs.on("open", tab => {` calls this also ` const userContextId = ContainerService._getUserContextIdFromTab(tab);`

Isn't this likely to cause the same issues with restyling? Assuming this was the actual bug you mentioned here: https://github.com/mozilla/testpilot-containers/issues/179#issuecomment-280594828

**Comment by bakulf** [comment]
The issue is about timing: We receive:
- open(UCI=0) - when the new tab is created after the dragging of the UCI=1 tab
- close(UCI=1) - the tab in the original window is closed
---- here the docShell is swapped between ----
- active(UCI=1) - finally we have something useful to use

File: index.js [hunk 1]
```diff
@@ -804,6 +811,8 @@ const ContainerService = {
       const indicator = viewFor(tab.window).document.getElementById("userContext-indicator");
       indicator.setAttribute("data-identity-icon", identity.image);
       indicator.style.listStyleImage = "";
+    }).then(() => {
```
Comment (bakulf) [inline index.js]: This is not part of the commit. sorry.

Comment (jonathanKingston) [inline index.js]: P sure this is in master from the previous merge no worries.

**Comment by jonathanKingston** [comment]
Cool LGTM then thanks for explaining 👍

**Comment by jonathanKingston** [comment]
@bakulf I'm afk from now though sorry didn't sign off irc
